File: samples/c/sqrt/sqrt.c
#include <stdio.h>
#include <stdlib.h>


double findSqrt(double number) {
    double x = number;
    double y = 1.0;
    const double e = 0.0000001;
    while (x - y > e) {
        x = (x + y)/2;
        y = number / x;
    }
    return x;
}


int main(int argc, char** argv) {
    if (argc != 2) {
        printf("Error: Usage %s <number>\n", argv[0]);
        return 1;
    }
    char* eptr;
    double number = strtod(argv[1], &eptr);


    if (*eptr != 0) {
        printf("Error: Expected to receive a number, recieved %s\n", argv[1]);
        return 2;
    }
    printf("%f\n", findSqrt(number));
}

File: samples/php/sqrt/sqrt.php
<?php

define("ACCURACY", 0.00000001);


function findSqrt($number) {
    $x = $number;
    $y = 1;
    while ($x - $y > ACCURACY) {
        $x = ($x + $y) / 2;
        $y = $number / $x;
    }
    return $x;
}

if ($argc != 2) {
    fwrite(STDERR, sprintf("Error: Usage ./%s <number>\n", $argv[0]));
    exit(1);
}

$number = floatval($argv[1]);
if (strval($number) != $argv[1]) {
    fwrite(STDERR, sprintf("Error: Expected to receive a number. Received '%s'\n", $argv[1]));
    exit(2);
}

echo sprintf("%f\n", findSqrt($number));

?>
